### ISSUE TITLE
adapt-contrib-slider Missing label for slider control

### ISSUE THREAD

**Issue by joe-replin** [adaptlearning/adapt-contrib-slider#156]
## Bug Description:
The Slider control does not have a label associated to indicate the purpose of the slider. This applies to the following components:

    Slider
    Confidence Slider 

## Steps to Reproduce:

    Using a screen reader, focus on the slider control, and note any label for the control. 

## Actual Behavior:

See bug description.

## Expected Behavior:

Apply an aria-label or aria-labelledby property to the slider control to indicate the purpose of the control (for example, associate the question with the slider).

**Comment by chris-steele** [comment]
@swashbuck is this something you can pick up with your React conversion?

**Comment by kirsty-hames** [comment]
I've had the same issue reported from an external auditor. Their suggested fix is to use `aria-describedby` to associate the question text e.g. `body` with the slider control. 

I think `aria-describedby` would be more suitable here as is used to reference longer content that provides a description. `aria-labelledby` should reference brief text that provides the element with an accessible name.

**Comment by joe-replin** [comment]
@kirsty-hames & @chris-steele, I would add aria-describedby to associate the question text with the slider control but we have the ariaQuestion field now being read at the '.component__widget' level. What makes better sense here?
- Move the ariaQuestion read-out to be on the slider control instead of at the widget level
- Keep the ariaQuestion read-out at the widget level and add an additional described-by reference on the slider control
- Leave it be and let the ariaQuestion read-out be the best descriptor?

Please reference the latest addition here: https://github.com/adaptlearning/adapt-contrib-slider/blob/865f71a4999639489e6c25080741fb5180c91dae/templates/slider.jsx#L37

**Comment by oliverfoster** [comment]
Keep the ariaQuestion solution, it's important to have a uniform way of doing this across the components.

**Comment by joe-replin** [comment]
@kirsty-hames, assigning this to you to determine if the current implementation with ariaQuestion would satisfy the issue your external auditor raised.

**Comment by kirsty-hames** [comment]
> @kirsty-hames, assigning this to you to determine if the current implementation with ariaQuestion would satisfy the issue your external auditor raised.

I think ariaQuestion is separate to the [issue raised by the external auditor](https://github.com/adaptlearning/adapt-contrib-confidenceSlider/issues/82) and just referring back to your query above...

> @kirsty-hames & @chris-steele, I would add aria-describedby to associate the question text with the slider control but we have the ariaQuestion field now being read at the '.component__widget' level. What makes better sense here?
> 
> * Move the ariaQuestion read-out to be on the slider control instead of at the widget level
> * Keep the ariaQuestion read-out at the widget level and add an additional described-by reference on the slider control
> * Leave it be and let the ariaQuestion read-out be the best descriptor?

I think we should keep the ariaQuestion read-out at the widget level (for consistency with other components) and add an additional described-by reference on the slider control to give context. I think `slider__label-start` and `slider__label-end` should be `aria-hidden` as these don't give any context stand alone, only when they are associated with the slider values.
